### PR TITLE
Api endpoints

### PR DESCRIPTION
**Pull request by Muzammilshabbir**
## Summary by CodeRabbit

- **New Features**
  - Added support for sharing files with client groups and managing shared files via new API endpoints.
  - Introduced client billing preferences management, allowing creation and update of billing preferences for client groups.
  - Enabled updating client portal permissions through a dedicated API endpoint.
  - Implemented unified billing document export functionality with detailed success/failure reporting.
  - Added support for multiple appointments per superbill and enhanced statement item management.
  - Introduced persistent storage and detailed formatting of statement items.
  - Added DELETE endpoint for superbills to support deletion and cleanup of related appointments.

- **Enhancements**
  - Refactored clinical information, practice information, and license management to operate directly on clinician entities.
  - Improved phone number validation across profile security, practice info, and security info forms, including real-time validation and error messaging.
  - Updated invoice number generation to use sequential numbering instead of timestamp-based.
  - Enhanced practice phone form with real-time validation and error messaging.
  - Improved billing documents API with optimized raw SQL queries, added support for receipts, and pagination.

- **Bug Fixes**
  - Improved error handling and validation for API endpoints, including more descriptive error messages and status codes.
  - Fixed validation and authorization edge cases across multiple API endpoints.

- **Database & Schema**
  - Major schema redesign introducing new tables for appointment notes, billing preferences, client files, group services, diagnoses, treatment plans, good faith estimates, and statement items.
  - Modified relationships and fields in existing tables to support new features, including clinician-centric data structures and export tracking.
  - Added and updated foreign key constraints and indexes for improved data integrity and performance.
  - Changed license ID from integer to UUID and associated licenses with clinicians instead of clinical info.

- **Tests**
  - Expanded and refactored test coverage to align with new data models and API logic, including edge cases and error scenarios.
  - Added new test suites for client billing preferences and portal permissions.
  - Updated existing tests to reflect clinician-centric data models and multi-appointment superbills.
  - Extended clinician-related test data to include new properties: speciality, NPI number, and taxonomy code.
  - Updated billing documents and invoice tests to include new fields and data structures.
  - Refactored statement and superbill tests to align with new relational data structures and API handlers.

- **Chores**
  - Updated internal data models and code structure to support new features and improved maintainability.

### CODE REVIEW

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

File: apps/back-office/__tests__/api/appointment/route.unit.test.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { vi } from "vitest";
 import { describe, it, expect, beforeEach } from "vitest";
 import { createRequest, createRequestWithBody } from "@mcw/utils";
-import { GET, POST, PUT, DELETE } from "@/api/appointment/route";
+import { GET, POST, PUT, DELETE } from "../../../src/app/api/appointment/route";
```
[the remark this answers is not in the document]
Comment (talhabalaj) [reply apps/back-office/__tests__/api/appointment/route.unit.test.ts:4]: Fix this please, @Muzammilshabbir

File: apps/back-office/__tests__/api/billing-documents/route.integration.test.ts [hunk 2]
```diff
@@ -16,15 +16,17 @@ vi.mock("@/utils/helpers", () => ({
 }));
 
 // Import the route handler AFTER mocks are defined
-import { GET } from "@/api/billing-documents/route";
+import { GET } from "../../../src/app/api/billing-documents/route";
```
Comment (talhabalaj) [inline apps/back-office/__tests__/api/billing-documents/route.integration.test.ts:19]: Make sure to fix this problem iin the whole code base.

File: apps/back-office/__tests__/api/billing-documents/route.unit.test.ts [hunk 3]
```diff
@@ -25,36 +26,82 @@ interface BillingDocument {
...
 const mockSuperbill = (overrides = {}) => {
-  return dataSuperbillMock(overrides);
+  const superbill = dataSuperbillMock(overrides);
+  return {
+    ...superbill,
+    is_exported: false,
+  };
 };
 
 const mockStatement = (overrides = {}) => {
```
Comment (talhabalaj) [inline apps/back-office/__tests__/api/billing-documents/route.unit.test.ts:51]: Please use mock from the mock factory we have.

File: apps/back-office/__tests__/api/billing-documents/route.unit.test.ts [hunk 4]
```diff
@@ -25,36 +26,82 @@ interface BillingDocument {
...
+  ];
+  return docs;
 };
 
 describe("Billing Documents API", () => {
   beforeEach(() => {
     vi.resetAllMocks();
 
+    // Mock $queryRawUnsafe which is used in the new implementation
+    prismaMock.$queryRawUnsafe = vi.fn() as any;
```
Comment (talhabalaj) [inline apps/back-office/__tests__/api/billing-documents/route.unit.test.ts:103]: Why are we using queryRawUnsafe? please use queryRaw only.

**Review by talhabalaj — APPROVED** [review]

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.

**Comment by ShawaizSarfraz117** [comment]
## 🚀 PR Deployment to QA
  
  This PR has been deployed to QA slots for testing:
  
  - **Back Office QA:** [https://back-office-mcw-dev-qa.azurewebsites.net](https://back-office-mcw-dev-qa.azurewebsites.net)
  - **Front Office QA:** [https://front-office-mcw-dev-qa.azurewebsites.net](https://front-office-mcw-dev-qa.azurewebsites.net)
  
  Please verify your changes in these environments before merging.
